### PR TITLE
MET-1334: Tiny change to add capability of making a filed to be not indexed

### PR DESCRIPTION
**Pull request by CCheSumo**
Tiny change to add capability of making a filed to be not indexed. Also updated version numbers.

> The index attribute controls how the string will be indexed. It can contain one of three values:
> analyzed
> not_analyzed
> no

### CODE REVIEW

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala [hunk 1]
```diff
@@ -39,11 +39,15 @@ trait MappingDsl extends DslCommons {
   sealed trait IndexType {
     val rep: String
   }
-
+  
   case object NotAnalyzedIndex extends IndexType {
     val rep = "not_analyzed"
   }
 
+  case object NotIndexedIndex extends IndexType {
```
Comment (rcoh) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala:47]: A comment explaining why someone might want this seemingly nonsensical mode or a link to the docs could be useful.

Comment (CCheSumo) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala:47]: We would like to maintain certain fields unindexed such that it is available when we retrieve the document, but do not take space and cpu when indexing.

Comment (davidandrzej) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala:47]: Could have some in-code commenting explaining what these mean and why you'd use them.

Comment (rcoh) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala:47]: Yeah what @davidandrzej said was what I meant ;-)

**Pull request merged**